arch/arm/soc/atmel_sam0/samd/Kconfig.soc

```diff
@@ -0,0 +1,89 @@
+# Kconfig - Atmel SAMD MCU series
+#
+# Copyright (c) 2017 Google LLC.
+# SPDX-License-Identifier: Apache-2.0
+
+choice
+	prompt "Atmel SAMD MCU Selection"
+	depends on SOC_SERIES_SAMD
+
+	config SOC_PART_NUMBER_SAMD21E15A
+		bool "SAMD21E15A"
+
+	config SOC_PART_NUMBER_SAMD21E16A
+		bool "SAMD21E16A"
+
+	config SOC_PART_NUMBER_SAMD21E17A
+		bool "SAMD21E17A"
+
+	config SOC_PART_NUMBER_SAMD21E18A
+		bool "SAMD21E18A"
+
+	config SOC_PART_NUMBER_SAMD21G15A
+		bool "SAMD21G15A"
+
+	config SOC_PART_NUMBER_SAMD21G16A
+		bool "SAMD21G16A"
+
+	config SOC_PART_NUMBER_SAMD21G17A
+		bool "SAMD21G17A"
+
+	config SOC_PART_NUMBER_SAMD21G17AU
+		bool "SAMD21G17AU"
+
+	config SOC_PART_NUMBER_SAMD21G18A
+		bool "SAMD21G18A"
+
+	config SOC_PART_NUMBER_SAMD21G18AU
+		bool "SAMD21G18AU"
+
+	config SOC_PART_NUMBER_SAMD21J15A
+		bool "SAMD21J15A"
+
+	config SOC_PART_NUMBER_SAMD21J16A
+		bool "SAMD21J16A"
+
+	config SOC_PART_NUMBER_SAMD21J17A
+		bool "SAMD21J17A"
+
+	config SOC_PART_NUMBER_SAMD21J18A
+		bool "SAMD21J18A"
+
+endchoice
+
+if SOC_SERIES_SAMD
+
+config SOC_ATMEL_SAMD_XOSC32K
+	bool "Enable the external 32 kHz crystal oscillator"
+	default y
```

Comment:
> and the three boards that I spot-checked all use XOSC32K so I made it the default clock source.

I still think it would be better to keep this setting board specific, i.e. set the default here to 'n', but I'm not insisting.

Reply:
Done.